### DISCUSSION TITLE
Sphinx build exception: HtmlFormatter object has no attribute 'get_linenos_style_defs'

### DISCUSSION THREAD

**Discussion by bocklund** [pradyunsg/furo#308] [Potential Issue] [answered]
In pycalphad's CI docs builds, I am getting a new (to us) error raised in Furo, but I don't know if this is a Furo issue or upstream issue.

In this [CI build], I get the error on a Sphinx build:

```
Extension error (furo):
Handler <function _overwrite_pygments_css at 0x7faf5a9329d0> for event 'build-finished' threw an exception (exception: 'HtmlFormatter' object has no attribute 'get_linenos_style_defs')
```

In our Sphinx config, we are using:

```python
pygments_style = 'default'
pygments_dark_style = "native"
```

In our previous successful build for the open PR ([from 2021-11-08](https://github.com/pycalphad/pycalphad/runs/4143438763?check_suite_focus=true)) we had the following requirements (via pip in the `install from requirements-dev.txt` step):

- `furo=2021.10.9`
- `sphinx=4.2.0`
- `docutils=0.17.1`

In the failing build, the installed versions  in [that CI build](https://github.com/pycalphad/pycalphad/actions/runs/1470193365) are:

- `furo=2021.11.16`
- `sphinx=4.3.0`
- `docutils=0.17.1`

We didn't make any changes to the docs between the successful and unsuccessful runs that could cause this error.  Some other non-docs, package-specific dependencies have changed, but they should be unrelated to this. 

Let me know if there's anything that I could do to further reproduce or narrow down the issue.

**Comment by pradyunsg** [comment]
A smaller self-contained reproducer would be great.

I am unable to reproduce this failure. I just tried building Furo's own docs against 4.3.0 (which, looks like I already had installed). First without the pygments configuration variables and once after, both builds were done with `-a` which makes Sphinx rebuild everything.

**Reply by pradyunsg** [reply]
Ah, you've got pygments 2.3.1 -- and I'm using an attribute from 2.7.0+.

**Comment by pradyunsg** [comment]
Should be fixed by bdcf32eea8fcdcc2a060dc88f03b2ac4041cff9a.

**Reply by bocklund** [reply]
Thank you!

For anyone else running into this, releases `furo<=2021.10.09` or (after the fix) `furo>2021.11.16` should work.

**Comment by chenxuuu** [comment]
same issue here, but I found that unless I install pygments==2.11.2 manually, it still give me this error

anyway, this problem is solved

**Comment by rdeveen** [comment]
I have the same problem, 
Furo version > 2022.6.4.1 failed. 
Version Furo == 2022.4.7 does work.

**Reply by rdeveen** [reply]
Can it be this change: https://github.com/pradyunsg/furo/commit/18374b79c62b2c5d092d0464ff0f699c52c031e6

**Reply by pradyunsg** [reply]
Whelp. This looks like I made a mistake!

**Comment by ubaldot** [comment]
I am running in the same problem. I have `furo 2022.6.21` installed through `conda-forge `.
How to fix it?

**Reply by pradyunsg** [reply]
Upgrade `pygments` and that should fix this.

**Comment by pradyunsg** [comment, marked as answer]
Fixed again, in 2022.09.15 – Pragmatic Pistachio.

**Reply by PttCodingMan** [reply]
Collecting furo
    Downloading furo-2022.9.15-py3-none-any.whl (326 kB)

Extension error (furo):
  Handler <function _overwrite_pygments_css at 0x7fa338b5c790> for event 'build-finished' threw an exception (exception: 'HtmlFormatter' object has no attribute 'get_linenos_style_defs')

still happen in version 2022.9.15.

**Reply by pradyunsg** [reply]
Looking through the discussions, no one has reported this since so I'm gonna go ahead and say this is resolved. :)

Hopefully you were able to resolve this. If not, please open a new discussion! :)

**Reply by ubaldot** [reply]
Thanks! I added a rocket reaction plus marked as an answer but I didn’t know how to close it at that time. :)

**Reply by pradyunsg** [reply]
No worries! :)
